src/lsp/cobol_lsp/lsp_completion.ml

```diff
@@ -48,22 +48,28 @@ let change ~(case: actual_case) = match case with
   | Uppercase -> String.uppercase
   | Lowercase -> String.lowercase
 
-let to_string qualnames =
-  List.flatten @@ List.rev_map (function
-      | Cobol_ptree.Name name -> [~&name]
-      | Qual (name,_) as qualname ->
-        [~&name; Pretty.to_string "%a" Cobol_ptree.pp_qualname qualname]
-      | _ -> []) qualnames
-
-let qualnames_proposal ~filename pos group =
-  match Lsp_lookup.cobol_unit_at_position ~filename pos group with
+let to_string = function
+  | Cobol_ptree.Name name -> [~&name]
+  | Qual (name,_) as qualname ->
+    [~&name; Pretty.to_string "%a" Cobol_ptree.pp_qualname qualname]
+  | _ -> []
+
+let qualnames_proposal ~filename pos group : (string * bool) list =
+  let comp_categories = Lsp_comp_semantic.type_at_position ~filename pos group in
```

Comment:
May we just move the contents of `lsp_comp_semantic.ml` so that reads `Lsp_lookup.type_at_position`? Could also be `Lsp_lookup_type.at_position`.